File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> sums = new HashMap<>();

        for (int i = 0; i < nums.length; i++)
        {
            sums.put(nums[i], i);
        }

        for (int i = 0; i < nums.length; i++)
        {
            int diff = target - nums[i];
            if (sums.containsKey(diff) && sums.get(diff) != i)
            {
                return new int[]{i, sums.get(diff)};
            }
        }

        return new int[]{};
    }
}

File: Data Structures & Algorithms/sort-array-by-parity/submission-1.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        n = len(nums)
        i, j, k = 0, 0, n // 2
        result = [0] * n

        while i < n // 2 and k < n:
            j = i
            while j < n:
                if j < n and nums[j] % 2 == 0:
                    result[i] = nums[j]
                    i += 1
                elif k < n:
                    result[k] = nums[j]
                    k += 1
                j += 1

        return result